### DISCUSSION TITLE
Valetudo RE and Homebridge

### DISCUSSION THREAD

**Discussion by limeos** [rand256/valetudo#382] [Q&A]
I'm running Valetudo RE V0.10.0 on my Roborock S50 with Firmware 3.5.8_002034. 
I found [this project](https://github.com/onfoot/homebridge-valetudo-xiaomi-vacuum) on github and wanted to install it.
Normally this should be as simple as `npm -g install homebridge-valetudo-xiaomi-vacuum` but it seems that the os on the vacuum does not even has a package manager like dpkg or apt installed. So my question is if there is any possibility to get homebridge-valetudo-xiaomi-vacuum installed on my current system. 

I would be happy for any comment.
Thanks.

**Comment by rand256** [comment]
As far as I've read just now from that project's README, you are supposed to install that on some other server - not on the vacuum itself.

Also it's meant for vanilla valetudo, not RE. So while something would work, some other functions in theory may require changing the sources to match RE's [REST API](https://github.com/rand256/valetudo/wiki/REST-API).

**Reply by limeos** [reply]
Thank you. I had read that too, but thought that in theory there should be nothing against running the homebridge on the vacuum as well (until I noticed that it's a very stripped down Linux on the vacuum and installation doesn't seem to be straightforward). I'm basically just interested in starting the standard cleaning from the home app or with Siri. Is that a feature that will work through the existing API of Valetudo RE? In my opinion, a homebridge integration would be an interesting feature for Valetudo RE. This would then also make it possible to control the vacuum via e.g. Alexa or Google Home.

**Reply by rand256** [reply]
Well, actually you can download a compiled node.js release for armv7, unpack it somewhere at `/mnt/data/<some new directory>` and do something like [this](https://github.com/nodejs/help/wiki/Installation). Most likely it will run. But yeah, it won't be very straightforward nor fast.

**Comment by limeos** [comment]
In case someone should stumble across this question again:
Meanwhile `homebridge-valetudo-xiaomi-vacuum` supports Valetudo RE. 
It is only important to set `"legacy-mode": true` in the configuration. 
Link to the project: https://github.com/onfoot/homebridge-valetudo-xiaomi-vacuum
